### ISSUE TITLE
Fake source code

### ISSUE THREAD

**Issue by grepwood** [SeriousCache/UABE#392]
https://github.com/DerPopo/UABE/archive/2.2stabled.tar.gz is not a link to the source code but instead to some sad readme files.

**Comment by grepwood** [comment]
From duckduckgo.com on github.com:
```
GitHub is where over 56 million developers shape the future of software, together. Contribute to the open source community, manage your Git repositories, review code like a pro, track bugs and features, power your CI/CD and DevOps workflows, and secure code before you commit it.
```

Since this project is not open source, it does not belong on GitHub.

**Comment by synapt** [comment]
Your quotation of the github website description does not in any form imply that "it does not belong on github".  Not is being fully open source a requirement of github, and their community guidelines had you bothered to read them even explicitly acknowledge that they host many closed-source projects.

That said, the link you are providing is automatically generated by github for releases, it's an automatic .zip of the current master repository (or relevant tag branch).  Had you bothered to look for even a brief moment you would have noticed there is also a manually uploaded zip called "UABE_API_2.2stabled.zip" (and similar in previous releases) which include the source code.  And that there is also a pending PR including the zip code (assumed likely from the zip).

TL;DR if you're interested in the source, it's in here; https://github.com/DerPopo/UABE/releases/download/2.2stabled/UABE_API_2.2stabled.zip

**Comment by xtremeqg** [comment]
He's got a point. You are of course totally free to use Github as you please, but what you are doing now is not really in the spirit of open source. There is not even any source code in this repository, despite someone else being kind enough to try and do something about it (#360).

I stumbled on this repository while trying to find some answers as to why my existing unpacker doesn't work for some newer archive version. Various other places places like https://github.com/imadr/Unity-game-hacking, https://github.com/RaduMC/AssetStudio and http://wiki.xentax.com/index.php/Unity_Assets offer little help. Those projects / documents are either abandoned, out of date, plain wrong or a combination of all three.

Unfortunately, while this may be the last remaining active project. There are no answers here. I've read through the "source" you mentioned, but they are just header files for a Windows-only dynamic library that has no documentation. You could say "yea just load up Visual Studio and compile away"... if I wasn't a Linux developer, sure. That's probably what I would try.

The point of open source is to share knowledge and experience, so that everyone can learn from each other. In addition, others will be able to assist and speed up development or make improvements. None of which is currently possible since there is no source whatsoever. There is nothing here to learn, nor to contribute to.

One of the biggest strong points of open source for me personally is that software can be made to run on any platform. Source code can be translated to different languages as needed. This is impossible with binary blobs.

Looks like I'll just have to reverse engineer my way through like before. Disappointing.

**Comment by grepwood** [comment]
As of now I'm trying to work with newer versions of Unity assets from Phasmophobia. It's only a strike of luck that this tool is still able to list the contents. That's about it for this tool's capabilities under Wine - it will crash when trying to extract the assets or work with plugins. I'm not switching to Windows just to extract some ghost sounds, and I'm feeling like doing the worst childhood chores when I need to start a VM to do simple things like this.

Thanks to the repo owner's consistent lack of commitment to free, libre and open source software, which GitHub is made for specifically according to its own description, nobody is able to correct these issues until the grey eminence or other source code hoarder will find spare time and motivation to do it themselves. It's understandable that you may not have time to do it now. Life, education, work and hobbies are important things and nobody can expect others to drop priorities. But if someone else wants to fix and improve your software for free, why would you stop them?

My advice is stop dragging your feet unless you want to wait for someone determined to remake this tool as free software from the ground-up and legitimately steal your swag. Nobody in their right mind is going to laugh at your source code's quality. It's a very human thing to make mistakes.

**Comment by grepwood** [comment]
I don't wanna be that guy who said I told ya, but myself and @xtremeqg told ya.
- https://github.com/DerPopo/UABE/issues/397
- https://github.com/DerPopo/UABE/issues/398
- https://github.com/DerPopo/UABE/issues/390
- https://github.com/DerPopo/UABE/issues/389
- https://github.com/DerPopo/UABE/issues/388

Keeps going on and on.

**Comment by grepwood** [comment]
@synapt I was unable to find this obscure link on my own. This is not a proper way of distributing source code - on GitHub and every version control system, source code must be in the repository, not in carefully zip files carefully hidden from public view. There is no open source project that practices this kind of lunacy.

Also, this is the source code for the API to interface with UABE, rather than for UABE itself. It has to be statically linked with some proprietary blobs from bin32 and bin64 directories.

I was also surprised to see the solution files are for VS2010. Someone's holding up for Windows XP support in this day and age?

**Comment by synapt** [comment]
@grepwood

Apologies I'm particularly way late on this reply as I got the notification ages ago and forgot about it till I just saw it again by chance.

I think the issue here is that you assume the source code zips would have the source code, and yes that would make sense except a bit of a critical issue; the reason you see others have the source code in there is because those two files are automatically generated by github and contain the at-release files of the repository branch being released, which should make it obvious why they wouldn't contain the source code in this case since the repository doesn't contain the source.  But there is nothing obscure about what I linked, it's literally there for every release on the releases page (which is very clearly available and visible from the project main page).  At minimum at any point you could have at least taken a moment of curiosity to open it and see, or discover it in a mere matter of minutes searching like I did.

That said, I don't actually entirely disagree with you, the source should be in the main public repository, what I do disagree with though is your very attitude from your first post and follow up is all bad attitude, particularly making things up with a contextual quote that didn't even actually state what you claimed.

More viable approaches would be to have asked why a pull request including the source hasn't been approved/merged yet, or even commenting on that pull request attempting to get DerPopo's attention.  Not just more viable, but more civil and, since you like to talk about respectable actions, also more respectable.

**Comment by JerryPLayz** [comment]
The problem is @synapt , that they call it an API. I don't believe they have documented at all. I want to integrate this into a project of mine, but I can't do that If I don't know what I'm working with. So not being able to see the source or some proper docs outlining key features is prohibitive to progress. It also doesn't help that I'm writing my app in C#, and yet the header is for C++. Similar, but for me, it means working externally... a painfully difficult process. (I don't mean to sound negative, it just means I need to find another way to extract unity assets... any ideas?)

**Comment by mskgroup** [comment]
[Here solved your problem#410](https://github.com/DerPopo/UABE/issues/410)

**Comment by brian799** [comment]
> Your quotation of the github website description does not in any form imply that "it does not belong on github". Being fully open source a not requirement of github, and their community guidelines had you bothered to read them even explicitly acknowledge that they host many closed-source projects.
> 
> That said, the link you are providing is automatically generated by github for releases, it's an automatic .zip of the current master repository (or relevant tag branch). Had you bothered to look for even a brief moment you would have noticed there is also a manually uploaded zip called "UABE_API_2.2stabled.zip" (and similar in previous releases) which include the source code. And that there is also a pending PR including the zip code (assumed likely from the zip).
> 
> TL;DR if you're interested in the source, it's in here; https://github.com/DerPopo/UABE/releases/download/2.2stabled/UABE_API_2.2stabled.zip

is there an example, how to use the API, before download it?

**Comment by LukeSaward1** [comment]
A repo does NOT need to have source code. wtf.

**Comment by grepwood** [comment]
That's not what GitHub, GitLab, Gogs and BitBucket were created for. This is for development of free software; free as in "AMERICA! F*** YEAH!"

You wanna do shady under-the-table software development, where nobody will criticise your source code or fork it to make better one? At least do it properly: use a combination of adf.ly links, lockerz.com links with zip passwords (or other pay-for-download sites), and use some obscure noname RapidShare clones that will probably expire within a year or two due to legal issues where you put the real artifacts inside an encrypted zip file, with the password you must first obtain from lockerz. That way you will ensure that once you, or any part of this convoluted artifact delivery flow is unreachable, your software is lost forever unless someone kindly paid all the middle-men when they were alive and out of prison and put it on archive.org.

**Comment by LukeSaward1** [comment]
Some repos don't contain any source code. Some are specifically for file hosting, for example.

**Comment by grepwood** [comment]
That's what RapidShare is for if they're still around anymore.

**Comment by LukeSaward1** [comment]
You can still host files on GitHub. That's what Project 1.27 does, that's what I do to host a copy of the game I'm writing a modloader for since I need a direct download to my zip file [had to put it in releases but still].
Also repos can be used for profile READMEs.

**Comment by LukeSaward1** [comment]
"Since this project is not open source, it does not belong on GitHub."

Ever heard of private repos?

**Comment by grepwood** [comment]
In my honest opinion that's abuse of publicly available free VCS. What you want to do is essentially handled by RapidShare, MegaUpload, GoogleDrive, etc. VCS is for version control where you can work on software in a collaborative way. Proprietary software limits this collaborative nature and GitHub has provisions for that, it's called a private repository. If you pony up for a GitHub subscription to set up private repos, or have enough moolah to set up a private instance of GitHub at your company, or you set up a private repository on BitBucket for free, or host your own instance of GitLab or Gogs, this is done properly - so you can work in a collaborative way on your software with a limited set of peers.

**Comment by LukeSaward1** [comment]
That's not abuse, what the fuck are you talking about?!

**Comment by synapt** [comment]
Why in the hell is this still going on?

The community guidelines were pointed to which clearly state this repo is fine to exist, period.  There is no argument or debate, opinions are irrelevant.  Anyone that doesn't like the community guideline, join github leadership and try to change it, otherwise good lord drop it lol.

**Comment by grepwood** [comment]
> That's not abuse, what the fuck are you talking about?!

You're trying to justify using a free VCS as some sort of free file hosting service. You're confusing GitHub with MegaUpload.

**Comment by LukeSaward1** [comment]
Shut the fuck up. I'm not justifying shit.
I'm stating a fact which your two-year-old brain can't understand.

**Comment by grepwood** [comment]
> Why in the hell is this still going on?
> 
> The community guidelines were pointed to which clearly state this repo is fine to exist, period. There is no argument or debate, opinions are irrelevant. Anyone that doesn't like the community guideline, join github leadership and try to change it, otherwise good lord drop it lol.

You dare me to get a job at Github just to get rid of repos like this? Maybe.

>I'm stating a fact which your two-year-old brain can't understand.
I'll make note of your ad-hominems and vulgar language. You don't seem to understand this is a version control system, not a free file hosting service like MegaUpload, DropBox, GoogleDrive or RapidShare. In case you still don't understand, a version control system is a service that offers tracking changes across source code files. Binaries aren't source code no matter how much time spent with HexRays IDA you throw at it.

**Comment by LukeSaward1** [comment]
If you can't accept that users are free to use GitHub in which ever way they like without violating GitHub TOS and community guidelines, then you need to get off of GitHub.

**Comment by grepwood** [comment]
> If you can't accept that users are free to use GitHub in which ever way they like without violating GitHub TOS and community guidelines, then you need to get off of GitHub.

And who are you to order people around, commanding others to leave GitHub if they do not agree with abusing GitHub?

**Comment by synapt** [comment]
> 
> You dare me to get a job at Github just to get rid of repos like this? Maybe.
> 

That is literally your only choice to get your way.  This repo is not violating anything, it is allowed to exist regardless of what you feel, that is all there needs to be said about it.  It has been over a year since your original griping, all you are doing is looking childish by claiming stuff that you were already shown to be wrong on.

You did not make github, you are not a github leader or executive, you have no position or grounds to deem what is abuse, the fact that it is explicitly covered within the guidelines clearly indicates it is not abuse, the only thing you have to agree on is the guidelines you clicked 'yes I agree' to when you made your account and continue using your account any time they change, that is the only agreement, there is no agreement of this being abuse because it is, in fact, by github guidelines, not abuse.

Has that clarified things?

**Comment by LukeSaward1** [comment]
Ask any GitHub admin/moderator if storing files in GitHub is abuse. They'll say it isn't. I hope someone comes across this issue and closes it, because this thread makes you (grepwood) come across as an asshole who advocates heavily for people not to use a feature of GitHub. Just because it's not designed for that doesn't mean that it should be used. It's a feature and it's intended. Get over it, grepwood.
